node/clients/masterdata.ts

```diff
@@ -0,0 +1,129 @@
+import { AuthenticationError, ForbiddenError, InstanceOptions, IOContext, JanusClient, RequestConfig, UserInputError } from '@vtex/api'
+import { AxiosError } from 'axios'
+import FormData from 'form-data'
+import { mergeAll, zipObj } from 'ramda'
+
+interface PaginationArgs {
+  page: number,
+  pageSize: number
+}
+
+interface Document {
+  Id: string
+  Href: string
+  DocumentId: string
+  [key: string]: any
+}
+
+interface KeyValue {
+  key: string
+  value: string
+}
+
+/*
+ * Convert a list of fields like [ {key: 'propertyName', value: 'String'}, ... ]
+ * to a JSON format.
+ */
+const parseFieldsToJson = (fields: KeyValue[]) => mergeAll(
+  fields.map((field: any) => zipObj([field.key], [field.value])),
+)
+
+const statusToError = (e: any) => {
+  if (!e.response) {
+    throw e
+  }
+  const { response } = e as AxiosError
+  const { status } = response!
+  if (status === 401) {
+    throw new AuthenticationError(e)
+  }
+  if (status === 403) {
+    throw new ForbiddenError(e)
+  }
+  if (status === 400) {
+    throw new UserInputError(e)
+  }
+  throw e
+}
+
+const paginationArgsToHeaders = ({page, pageSize}: PaginationArgs) => {
+  if (page < 1) {
+    throw new UserInputError('Smallest page value is 1')
+  }
+  const startIndex = (page - 1) * pageSize
+  const endIndex = startIndex + pageSize
+  return {
+    'REST-Range': `resources=${startIndex}-${endIndex}`
+  }
+}
+
+export class MasterData extends JanusClient {
+  constructor (ctx: IOContext, options?: InstanceOptions) {
+    super(ctx, {
+      ...options,
+      headers: {
+        ...options && options.headers,
+        ...{Accept: 'application/vnd.vtex.ds.v10+json'},
+        ...ctx.adminUserAuthToken ? {VtexIdclientAutCookie: ctx.adminUserAuthToken} : null,
+        ...ctx.storeUserAuthToken ? {[`VtexIdclientAutCookie_${ctx.account}`]: ctx.storeUserAuthToken} : null,
+      },
+    })
+  }
+
+  public getDocument = (acronym: string, id: string, fields: string[]) => this.get<Document>(
```

Comment:
The type on get is different.

See:

```js
{ id: '4fc44cc0-6876-11e9-8289-0a1e532cd784',
     dataEntityId: 'productreview',
     accountId: '7b6a2d3d-d59f-47a8-998d-af140fd1a1d0',
     accountName: 'storecomponents',
     followers: [],
     schemas: [],
     metric: 'masterdata-createDocument',
     params: { foo: '123' },
     createdBy: 'B5AA99DB-B2E3-4187-8A98-8C266386599C',
     createdBy_USER:
      { Id: 'B5AA99DB-B2E3-4187-8A98-8C266386599C',
        Login: 'breno@vtex.com.br',
        Name: null },
     createdIn: '2019-04-26T22:55:01.0360561Z',
     lastInteractionBy: 'B5AA99DB-B2E3-4187-8A98-8C266386599C',
     lastInteractionBy_USER:
      { Id: 'B5AA99DB-B2E3-4187-8A98-8C266386599C',
        Login: 'breno@vtex.com.br',
        Name: null },
     lastInteractionIn: '2019-04-26T22:55:01.0360561Z',
     tags: [],
     dataInstanceId: '4fc44cc0-6876-11e9-8289-0a1e532cd784' } }
```

Reply:
I left it typed as `any` because the returned keys differ depending on the content of the `fields` and `acronym` arguments